core/src/main/java/discord4j/core/object/entity/Message.java

```diff
@@ -235,49 +235,44 @@ public boolean mentionsEveryone() {
     }
 
     /**
-     * Gets the IDs of the users specifically mentioned in this message.
+     * Gets the IDs of the users specifically mentioned in this message, without duplication and with the same order
+     * as in the message.
      *
-     * @return The IDs of the users specifically mentioned in this message.
+     * @return The IDs of the users specifically mentioned in this message, without duplication and with the same order
+     * as in the message.
      */
-    public Set<Snowflake> getUserMentionIds() {
-        // TODO FIXME we throw away member data here
+    public List<Snowflake> getUserMentionIds() {
         return data.mentions().stream()
                 .map(UserData::id)
                 .map(Snowflake::of)
-                .collect(Collectors.toSet());
-    }
-
-    /**
-     * Requests to retrieve the users specifically mentioned in this message.
-     *
-     * @return A {@link Flux} that continually emits {@link User users} specifically mentioned in this message. If an
-     * error is received, it is emitted through the {@code Flux}.
-     */
-    public Flux<User> getUserMentions() {
-        return Flux.fromIterable(getUserMentionIds()).flatMap(gateway::getUserById);
+                .collect(Collectors.toCollection(LinkedList::new));
     }
 
     /**
-     * Requests to retrieve the users specifically mentioned in this message, using the given retrieval strategy.
+     * Gets the users specifically mentioned in this message, without duplication and with the same order
+     * as in the message.
      *
-     * @param retrievalStrategy the strategy to use to get the users
-     * @return A {@link Flux} that continually emits {@link User users} specifically mentioned in this message. If an
-     * error is received, it is emitted through the {@code Flux}.
+     * @return The users specifically mentioned in this message, without duplication and with the same order
+     * as in the message.
      */
-    public Flux<User> getUserMentions(EntityRetrievalStrategy retrievalStrategy) {
-        return Flux.fromIterable(getUserMentionIds())
-                .flatMap(id -> gateway.withRetrievalStrategy(retrievalStrategy).getUserById(id));
+    public List<User> getUserMentions() {
+        // TODO FIXME we throw away member data here
+        return data.mentions().stream()
+                .map(data -> new User(gateway, data))
+                .collect(Collectors.toCollection(LinkedList::new));
```

Comment:
Is there a reason to prefer this over `Collectors.toList()`?

Reply:
Same reason as using LinkedList for getUserMentionIds I think?

Reply:
Sorry, I meant to ask what that reason is (for all of them)

Reply:
Oh, I forgot that `List` are ordered, I change to `toList`

Reply:
Done